R/jjcsformats.R

```diff
@@ -0,0 +1,230 @@
+        )
+      }
+
+      #values <- Map(y = x, fun = roundings, function(y, fun) fun(y))
+      values2 <- list()
+      for (i in 1:length(x)){
+        values2[[i]] <- roundings[[i]](x[[i]],na_str=na_str[[i]])
+      }
+
+      regmatches(x = str, m = positions)[[1]] <- values2
+      return(str)
+    }
+    return(rtable_format)
+  }
+  return(fnct)
+}
+
+
+jjcsformat_xx_SAS <- jjcs_format_xx_fct(roundmethod="SAS")
+jjcsformat_xx_R <- jjcs_format_xx_fct(roundmethod="R")
+
+
+### if we ever decide to switch rounding method, we just have to update jjcsformat_xx here
+
+jjcsformat_xx <- jjcsformat_xx_SAS
+
+#' Title: Formatting count and fraction values
+#'
+#' @description
+#'
+#' jjcsformat_count_fraction_fct: Formatting function to construct formats for presenting a count together with fraction (and/or denominator) with special consideration when count is 0, or fraction is 1.
+#' \cr See also: {tern::format_count_fraction_fixed_dp()}
+#'
+#' @inheritParams jjcs_format_xx_fct
+#' @importFrom tidytlg roundSAS
+#' @param type One of count_fraction or count_denom_fraction
+#' @family JJCS formats
+#' @rdname Count_fraction
+#' @name Count_fraction
+#' @export
+#'
+#' @examples
+#' jjcsformat_count_fraction <- jjcsformat_count_fraction_fct("SAS","count_fraction")
+#' jjcsformat_count_denom_fraction <- jjcsformat_count_fraction_fct("SAS","count_denom_fraction")
+
+#' jjcsformat_count_fraction(c(7,0.7))
+#' jjcsformat_count_fraction(c(70000,0.9999999))
+#' jjcsformat_count_fraction(c(70000,1))
+#'
+#' jjcsformat_count_fraction_fct("SAS","count_denom_fraction")(c(3,2000,3/2000))
+#' jjcsformat_count_fraction_fct("R","count_denom_fraction")(c(3,2000,3/2000))
+
+jjcsformat_count_fraction_fct <- function(roundmethod=c("SAS","R"),type=c("count_fraction","count_denom_fraction")){
+
+  roundmethod <- match.arg(roundmethod)
+  type <- match.arg(type)
+
+  if (roundmethod == "SAS"){roundfun <- tidytlg::roundSAS}
+  if (roundmethod == "R"){roundfun <- round}
+
+
+  #' @param x `numeric`\cr with elements `num` and `fraction` or `num`, `denom` and `fraction`.
+  #' @param d numeric(1). Number of digits to round fraction to (default=1)
+  #' @return A string in the format `count / denom (ratio %)`. If `count` is 0, the format is `0`. If fraction is >0.99, the format is `count / denom (>99.9%)`
+
+  fun <- function(x,output,d=1){
+    checkmate::assert_vector(x)
+
+    count <- x[1]
+    checkmate::assert_integerish(count)
+
+    if (type == "count_fraction"){
+      denom <- NULL
+      fraction <- x[2]
+      fdenom <- NULL
+
+      checkmate::assert_vector(x,min.len=2,max.len=2)
+    }
+    if (type == "count_denom_fraction"){
+      denom <- x[2]
+      fraction <- x[3]
+      fdenom <- paste0("/",denom)
+      checkmate::assert_vector(x,min.len=3,max.len=3)
+    }
+
+    attr(x, "label") <- NULL
+    if (any(is.na(x))) {
+      return("-")
+    }
+
+    tern:::assert_proportion_value(fraction, include_boundaries = TRUE)
```

Comment:
do not use internals of other packages in general. In this case we absolutely want to avoid circular dependencies